splito/lohi/_lo.py

```diff
@@ -0,0 +1,160 @@
+from rdkit import Chem, DataStructs
+from rdkit.Chem import AllChem
+import numpy as np
+from tqdm import tqdm
+
+from ._utils import get_similar_mols
+
+class LoSplitter():
+    def __init__(self, threshold=0.4, min_cluster_size=5, max_clusters=50, std_threshold=0.60):
+        """
+        Parameters:
+            threshold -- molecules with a similarity higher than this value are considered similar.
+            min_cluster_size -- number of molecules per cluster.
+            max_clusters -- maximum number of selected clusters. The remaining molecules go to the training set.
+            std_threshold -- Lower bound of the acceptable standard deviation for a cluster. It should be greater than measurement noise.
+                            If you're using ChEMBL-like data, set it to 0.60 for logKi and 0.70 for logIC50.
+                            Set it lower if you have a high-quality dataset.
+        """
+        self.threshold = threshold
+        self.min_cluster_size = min_cluster_size
+        self.max_clusters = max_clusters
+        self.std_threshold = std_threshold
+
+    def split(self, smiles, values, verbose=1):
```

Comment:
A user might want to get multiple splits (e.g. to get an estimate of the variance for statistical testing). Is there a way to make this split yield multiple splits? Similar to how `scikit-learn` does this? 

```python
splitter = LoSplitter(n_splits=5)
for train, test in splitter.split(data):
    # do something
```
Or is the split deterministic?

Reply:
This is an unavoidable difference from the other splitters in Splito. The splitter is very deterministic. I tried to generate different splits for Lo-Hi benchmark, but they are so similar that I don't think it makes sense to consider variation. I think it's better not to mimic the sklearn generator interface, to make the deterministic nature straightforward for users.

Reply:
Clear! Maybe we can keep the interface the same, but throw an error when `n_splits > 1`? I think that would be a better user experience. What do you think?

Reply:
I see you want to keep the interfaces the same to make it easier to use different splitters. I think that's a good intention. However, note that even with this fake parameter, there are many differences:

The Lo splitter constructor has unusual parameters: threshold, min_cluster_size, max_clusters, and std_threshold.
The `split()` function takes not only SMILES but also values.

In this case, the Lo splitter simply can't have the same interface. I don't see from here if adding a fake n_splits would help users, but I'm not familiar with Splito. Let me know if you still think it would be better to add this parameter, and I'll do it. For example, we can add **kwargs and throw an error if it includes `n_splits`.

Reply:
Makes sense! Let's leave it as is for now!